src/transformer/events/mod_quiz/attempt_reviewed.php

```diff
@@ -9,7 +9,9 @@ function attempt_reviewed(array $config, \stdClass $event) {
     $learner = $repo->read_record_by_id('user', $event->relateduserid);
     $instructor = $repo->read_record_by_id('user', $event->userid);
     $course = $repo->read_record_by_id('course', $event->courseid);
-    $attempt = $repo->read_record_by_id('attempt', $event->objectid);
+    $attempt = $repo->read_record_by_id('quiz_attempts', $event->objectid);
+    // Quiz attempts don't have names, so this will resolve an issue with the batch send to the LRS later.
+    $attempt->name = 'attempt reviewed';
```

Comment:
Is the missing attempt name causing an issue on Line 48 in the `context.contextActivities.grouping`? If so I think a better solution would be to make a new util to replace line 48 with something like `utils\get_activity\attempt($config, $attempt, $lang)`. This could then be reused elsewhere to fix the issue in a single function in all uses. Hope that makes sense.

Reply:
Yea! That was definitely a fix to get me along to the next issue. Till I found a better way to do it. I will make another util and add it where the $attempt->name is called, I believe it is really just that one return. In which case I'll add the function call there and have a function with a check to see if name is set and if not run it through a switch statement to find the best name depending on the activity table. Shouldn't be too hard and should be much cleaner then what I'm doing now. I'll hit it Monday, if the urge doesn't strike over the weekend.

Reply:
No worries, thought as much. Okay sounds good 👍